app/test_main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+from unittest.mock import patch
+
+from app.main import can_access_google_page
+
+
+def test_cannot_access_if_only_valid_url() -> None:
+    url = "https://www.google.com"
+
+    with patch("app.main.valid_google_url", return_value=True), \
+         patch("app.main.has_internet_connection", return_value=False):
+        result = can_access_google_page(url)
+        assert result == "Not accessible"
+
+
+def test_can_access_google_page_invalid_url() -> None:
```

Comment:
The test name `test_can_access_google_page_invalid_url` is a bit misleading. It might be better to rename it to `test_cannot_access_if_url_is_invalid` to reflect that the test checks the scenario where the URL is invalid but there is an internet connection.